packages/nimble-components/src/table-column/text-base/cell-view/index.ts

```diff
@@ -0,0 +1,32 @@
+import { observable, volatile } from '@microsoft/fast-element';
+import { TableCellView } from '../../base/cell-view';
+import type { TableCellRecord } from '../../base/types';
+
+/**
+ * The cell view base class for displaying fields of any type as text.
+ */
+export abstract class TableColumnTextCellViewBase<
+    TCellRecord extends TableCellRecord = TableCellRecord,
+    TColumnConfig = unknown
+> extends TableCellView<TCellRecord, TColumnConfig> {
+    /** @internal */
+    @observable
+    public isValidContentAndHasOverflow = false;
+
+    /** @internal */
+    public textSpan!: HTMLElement;
+
+    /** Returns the text to render in the cell when it contains a valid value (i.e. when shouldUsePlaceholder() is false) */
+    public abstract get text(): string;
+
+    /** Returns the text to render in the cell when it contains an invalid value (i.e. when shouldUsePlaceholder() is true) */
+    public abstract get placeholder(): string;
+
+    /** Return whether to display the placeholder value or the text value */
+    public abstract get shouldUsePlaceholder(): boolean;
+
```

Comment:
I guess I'm not sure what the reasoning is behind changing away from the existing implementation of the `content` property (and exposing what I think are redundant properties of `text` and `placeholder`). The `@volatile` nature of the previous `content` implementation worked because both `columnConfig` and `cellRecord` were/are `@observable` properties on `TableCellView` (and whose instances were immutable in practice).

Why isn't the implementation of the `text` property _always_ a mirror of the `value` property in the `TableCellRecord` given to the `TableCell`? Similar question for the `placeholder`. I would think this would _always_ be (regardless of the concrete extension of `TableColumnTextCellViewBase`) a reflection of the `placeholder` property available via `TCellConfig`, which I think in this class could be an extension of `TableColumnWithPlaceholderColumnConfig`.

Reply:
> Why isn't the implementation of the text property always a mirror of the value property

So the column can define a custom formatter, ie.e based on columnConfig render number `12` as `$12`.
This is saying a numeric column would implement the `get text` accessor and it would then access CellRecord value and columnconfig to provide the formatted `text` property to the template

Reply:
Ah that makes sense.

Reply:
At the moment I think I prefer Milan's first option he outlines, where we just document that a derived implementation _must_ mark the implemented abstract members as `@volatile`. Creating more implementation rigor around this seems like we're trying to protect ourselves from unlikely corner-cases, particularly for the rare situation where a client team wants to create their own column implementation.

Reply:
For now I went with option 1. I added a note to the task for creating client documentation of custom text column APIs to remind us to document the `@volatile` guidance in more detail. Hopefully by the time we do that we have a few more column types under our belt and have developed more crisp guidance.